### DISCUSSION TITLE
wrap-ansi / cliui issue unable to use the cli to do migrations / schema updates

### DISCUSSION THREAD

**Discussion by stearnsbq** [mikro-orm/mikro-orm#3029] [Q&A]
**Describe the bug**
Unable to run the cli using ```npx mikro-orm schema:update --run``` command, seems to be an issue with cliui's dependency on wrap-ansi causing issues on windows. I have tried installing the required packages to use the cli both globally and locally and same issue arises, also tried updating node to v17, tried just updating cliui and wrap-ansi but ran into commonjs vs es6 import issues, tried making a fresh project same issue. 

**Stack trace**
```
....\node_modules\cliui\node_modules\wrap-ansi\index.js:164
                const code = ansiStyles.codes.get(Number(escapeCode));
                                ^
TypeError: Cannot read properties of undefined (reading 'get')
    at exec (....\node_modules\cliui\node_modules\wrap-ansi\index.js:164:33)
    at ....\node_modules\cliui\node_modules\wrap-ansi\index.js:184:16
    at Array.map (<anonymous>)
    at module.exports (....\node_modules\cliui\node_modules\wrap-ansi\index.js:184:4)
    at ....\node_modules\cliui\index.js:210:19
    at Array.forEach (<anonymous>)
    at UI._rasterize (....\node_modules\cliui\index.js:206:9)
    at UI.rowToString (....\node_modules\cliui\index.js:122:10)
    at ....\node_modules\cliui\index.js:110:12
    at Array.forEach (<anonymous>)
```

**To Reproduce**
1. Create a project using mikro-orm (postgres driver)
2. Try to run a schema update

**Expected behavior**
The CLI should update postgres with defined schema

**Versions**

| Dependency | Version |
| - | - |
| node | 16.14.2 |
| typescript | 4.5.5 |
| mikro-orm | 5.1.2 |
| postgres | 5.1.2 |

**Comment by B4nan** [comment]
> I have tried installing the required packages to use the cli both globally and locally

You need to install it locally, not globally.

>  tried just updating cliui and wrap-ansi but ran into commonjs vs es6 import issues, tried making a fresh project same issue.

`cliui` and `wrap-ansi` are transitive dependencies, we dont use any of those directly. You should not install those yourself, neither you should mangle with their versions. You definitely don't want to use latest version of `wrap-ansi` (v8) as that is ESM only and [is not what `yargs/cliui` expects](https://github.com/yargs/cliui/blob/master/package.json#L54).

**Reply by B4nan** [reply]
You almost certainly managed to install wrong version of `wrap-ansi`. The stack trace points to v6, as you can see the problematic line here: https://unpkg.com/browse/wrap-ansi@6.2.0/index.js

But yargs/cliui expects v7, that line that throws for you is located elsewhere in that version. Try wiping lock file, remove the global installs, node_modules, and reinstall.

**Reply by stearnsbq** [reply]
Thanks for you reply, I see that yes it's installing wrap-ansi version 6.2.0. I tried deleting global and local packages, then same issue arises.

Edit: I tried a fresh new project again and the cli run successfully, there must be a problem with my entities. Not sure where to debug this

**Reply by stearnsbq** [reply]
Ive decided just to go with the programmatic route of doing this thanks!
